xmtp_mls/src/storage/errors.rs

```diff
@@ -63,11 +63,12 @@ impl RetryableError for diesel::result::Error {
 impl RetryableError for StorageError {
     fn is_retryable(&self) -> bool {
         match self {
-            Self::DieselConnect(connection) => {
-                matches!(connection, diesel::ConnectionError::BadConnection(_))
-            }
+            Self::DieselConnect(_) => true,
             Self::DieselResult(result) => retryable!(result),
             Self::Pool(_) => true,
+            Self::Lock(_) => true,
+            Self::PoolNeedsConnection => true,
```

Comment:
PoolNeedsConn/SqlCipherNot loaded can't really be retryable here I don't think, since a SqlCipher error indicates a compile-time linking issue and PoolNeedsConnection would be an error on SDK side that can't be resolved from libxmtp

Also, my bad, I should be better with remembering to add these to the `retryable` 😓

Reply:
Thanks for clarifying. I misunderstood the sqlcipher error as something that could happen if the pragma statement didn’t apply. Will remove.